aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthService.kt

```diff
@@ -17,34 +17,62 @@ package com.amplifyframework.auth.cognito
 
 import aws.sdk.kotlin.services.cognitoidentity.CognitoIdentityClient
 import aws.sdk.kotlin.services.cognitoidentityprovider.CognitoIdentityProviderClient
+import aws.sdk.kotlin.runtime.http.operation.customUserAgentMetadata
+import aws.smithy.kotlin.runtime.client.RequestInterceptorContext
 import aws.smithy.kotlin.runtime.client.endpoints.Endpoint
 import aws.smithy.kotlin.runtime.client.endpoints.EndpointProvider
+import aws.smithy.kotlin.runtime.http.interceptors.HttpInterceptor
 import com.amplifyframework.statemachine.codegen.data.AuthConfiguration
 
 interface AWSCognitoAuthService {
     var cognitoIdentityProviderClient: CognitoIdentityProviderClient?
     var cognitoIdentityClient: CognitoIdentityClient?
+    var customUserAgentPairs: MutableMap<String, String>
```

Comment:
In fact they should all be `val`. Can we make that change? Then do this:

```kotlin
val customUserAgentPairs = mutableMapOf<String, String>()
val cognitoIdentityProviderClient =  // ...
val cognitoIdentityClient = // ...

return object : AWSCognitoAuthService {
    override val cognitoIdentityProviderClient = cognitoIdentityProviderClient
    override val cognitoIdentityClient = cognitoIdentityClient
    override val customUserAgentPairs = customUserAgentPairs
} 
```

Reply:
changed the map, but I believe val doesn’t work for `cognitoIdentityProviderClient` or `cognitoIdentityClient` due to the changes I made (it would’ve been possible before).  the `AWSCognitoAuthService` needs to be defined at the start of the function so that it can be referenced in the transformer, so there’s a sort of loop between the service and the two clients.  to break the loop, we create the service first and then reassign its clients